src/wallets/web3-provider/methods/eth_getTransactionCount.js

```diff
@@ -0,0 +1,45 @@
+import utils from 'web3-utils';
+import { toPayload } from './jsonrpc';
+import EthCalls from '../web3Calls';
+import store from 'store';
+
+export default async ({ payload, requestManager }, res, next) => {
+  if (payload.method !== 'eth_getTransactionCount') return next();
+  const ethCalls = new EthCalls(requestManager);
+  const addr = payload.params[0];
+  let storedNonce = 0;
+  let fetchedNonce;
+  if (store.get(utils.sha3(addr)) === undefined) {
+    store.set(utils.sha3(addr), {
+      nonce: storedNonce,
+      timestamp: +new Date()
+    });
+  } else {
+    storedNonce = store.get(utils.sha3(addr)).nonce;
+  }
+  const lastFetch =
+    Math.round(
+      (new Date().getTime() - store.get(utils.sha3(addr)).timestamp) / 1000
+    ) / 60; // Get minutes
+  if (lastFetch < 1) {
+    fetchedNonce = storedNonce;
+  } else {
+    fetchedNonce = await ethCalls.getTransactionCount(addr);
```

Comment:
You only save the fetchednonce if it is higher than the stored one, right now for example if the last checked is over 1 min, it'll replace the stored nonce with the fetchednonce which can result in transactions with the same nonce